core/src/rpc_subscriptions.rs

```diff
@@ -199,50 +206,62 @@ impl RpcNotifier {
 
 fn filter_account_result(
     result: Option<(Account, Slot)>,
-    root: Slot,
-) -> Box<dyn Iterator<Item = RpcAccount>> {
+    last_notified_slot: Slot,
+) -> (Box<dyn Iterator<Item = RpcAccount>>, Slot) {
     if let Some((account, fork)) = result {
-        if fork >= root {
-            return Box::new(iter::once(RpcAccount::encode(account)));
+        if fork > last_notified_slot {
```

Comment:
I think we should drop this check

Imagine this scenario:
Account modified in slot 10 and we subscribe at slot 20 (last notified set to 10)
Account changes in slot 21 and we notify, setting last notified to 21 and fire off a notification
Slot 21 ends up being a fork, and we process slot 22 (account last changed is back to slot 10)

I think we should notify on slot 22 because slot 10 != slot 21. Thoughts?

Reply:
As discussed, changing the check to an inequality check, sounds good!